libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -346,24 +346,19 @@ bigint StaticAnalyzer::structureSizeEstimate(Type const& _type, set<StructDefini
 	case Type::Category::Array:
 	{
 		auto const& t = dynamic_cast<ArrayType const&>(_type);
-		return structureSizeEstimate(*t.baseType(), _structsSeen) * (t.isDynamicallySized() ? 1 : t.length());
+		return t.isDynamicallySized() ? bigint(1) : (structureSizeEstimate(*t.baseType(), _structsSeen) * t.length());
```

Comment:
How about

```suggestion
		if (!t.isDynamicallySized())
			return structureSizeEstimate(*t.baseType(), _structsSeen) * t.length();
			
		break;
```

for consistency with the mapping case which also relies on the default?

Reply:
Done, d0deb56.